sway-lib-std/src/option.sw

```diff
@@ -1,18 +1,86 @@
-//! Error handling with the `Option` type.
+//! Optional values.
 //!
-//! [`Option<T>`][`Option`] is the type used for representing the existence or absence of a value. It is an enum with the variants, [`Some(T)`], representing
-//! some value, and [`None()`], representing
-//! no value.
+//! Type [`Option`] represents an optional value: every [`Option`]
+//! is either [`Some`] and contains a value, or [`None`], and
+//! does not. [`Option`] types are very common in Sway code, as
+//! they have a number of uses:
+//!
+//! * Initial values
+//! * Return value for otherwise reporting simple errors, where [`None`] is
+//!   returned on error
+//! * Optional struct fields
+//! * Optional function arguments
+//!
+//! [`Option`]s are commonly paired with pattern matching to query the presence
+//! of a value and take action, always accounting for the [`None`] case.
+//!
+//! ```
+//! fn divide(numerator: u64, denominator: u64) -> Option<u64> {
+//!     if denominator == 0 {
+//!         Option::None
+//!     } else {
+//!         Option::Some(numerator / denominator)
+//!     }
+//! }
+//!
+//! // The return value of the function is an option
```

Comment:
Maybe put the remaining block of code in its own function, like `call_divide` or something like that?